src/main/scala/chisel3/ltl/LTL.scala

```diff
@@ -414,8 +415,10 @@ sealed abstract class AssertPropertyLike {
   )(
     implicit sourceInfo: SourceInfo
   ): Unit = {
-    val clocked = clock.fold(prop)(prop.clock(_))
-    createIntrinsic(label)(sourceInfo)(clocked.inner, disable.map(!_.value))
+    block(layer) {
+      val clocked = clock.fold(prop)(prop.clock(_))
+      createIntrinsic(label)(sourceInfo)(clocked.inner, disable.map(!_.value))
+    }
```

Comment:
I think it's important to handle the case (and we should show this in a test) of what happens if a user has their own layer and then creates an `AssertProperty` (or friend). As we discussed offline, I think that this `layer` should be treated as a default that only applies if the user hasn't specified a layer. WDYT?

Regardless, we need a test for that case (and for the default cases, just checking the emitted CHIRRTL is fine).

Reply:
The `layer.block` API is extended to provide this behavior here: https://github.com/chipsalliance/chisel/pull/4323/commits/2b80245b484cbeab6d5c542c2fd8d632c7769564

The LTL ops are updated to use it here: https://github.com/chipsalliance/chisel/pull/4323/commits/3a57da039b59ef7cb9f2ba0f337dc656d48a2adf These will no longer be put into the `Verification.Assert` block if the user decides to put them in _another_ block first.